File: 백준/2164/code.cpp
#include <iostream>
#include <queue>

using namespace std;

int main(){
    int n;
    queue<int> q;
    int a;
    cin >> n;

    for (int i=1;i<=n;i++){
        q.push(i);
    }

    
    while (q.size() > 1){
        q.pop();
        a = q.front();
        q.pop();
        q.push(a);
    }
    a = q.front();
    cout << a;
}

File: leetcode/Implement Queue using Stacks/code.py
class MyQueue:

    def __init__(self):
        self.queue = []

    def push(self, x: int) -> None:      
        self.queue.insert(0,x)       
        
    def pop(self) -> int:
        length = len(self.queue)
        var = self.queue[-1]
        del self.queue[length-1]
        return var

    def peek(self) -> int:
        var = self.queue[-1]
        return var

    def empty(self) -> bool:
        if len(self.queue) == 0:
            return True
        return False
        


# Your MyQueue object will be instantiated and called as such:
# obj = MyQueue()
# obj.push(x)
# param_2 = obj.pop()
# param_3 = obj.peek()
# param_4 = obj.empty()